src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -1405,6 +1451,82 @@ void ObjectMgr::LoadEquipmentTemplates()
     LOG_INFO("server.loading", " ");
 }
 
+void ObjectMgr::LoadCreatureMovementOverrides()
+{
+    uint32 oldMSTime = getMSTime();
+
+    _creatureMovementOverrides.clear();
+
+    // Load the data from creature_movement_override and if NULL fallback to creature_template_movement
+    QueryResult result = WorldDatabase.Query("SELECT cmo.SpawnId,"
+                                             "COALESCE(cmo.Ground, ctm.Ground),"
+                                             "COALESCE(cmo.Swim, ctm.Swim),"
+                                             "COALESCE(cmo.Flight, ctm.Flight),"
+                                             "COALESCE(cmo.Rooted, ctm.Rooted),"
+                                             "COALESCE(cmo.Chase, ctm.Chase),"
+                                             "COALESCE(cmo.Random, ctm.Random),"
+                                             "COALESCE(cmo.InteractionPauseTimer, ctm.InteractionPauseTimer) "
+                                             "FROM creature_movement_override AS cmo "
+                                             "LEFT JOIN creature AS c ON c.guid = cmo.SpawnId "
+                                             "LEFT JOIN creature_template_movement AS ctm ON ctm.CreatureId = c.id");
+    if (!result)
+    {
+        LOG_INFO("server.loading", ">> Loaded 0 creature movement overrides. DB table `creature_movement_override` is empty!");
+        return;
+    }
+
+    do
+    {
+        Field*              fields  = result->Fetch();
+        ObjectGuid::LowType spawnId = fields[0].GetUInt32();
+        if (!GetCreatureData(spawnId))
+        {
+            LOG_ERROR("sql.sql", "Creature (GUID: %u) does not exist but has a record in `creature_movement_override`", spawnId);
+            continue;
+        }
+
+        CreatureMovementData& movement = _creatureMovementOverrides[spawnId];
+        if (!fields[1].IsNull())
+        {
+            movement.Ground = static_cast<CreatureGroundMovementType>(fields[1].GetUInt8());
+        }
+
+        if (!fields[2].IsNull())
+        {
+            movement.Swim = fields[2].GetBool();
+        }
+
+        if (!fields[3].IsNull())
+        {
+            movement.Flight = static_cast<CreatureFlightMovementType>(fields[3].GetUInt8());
+        }
+
+        if (!fields[4].IsNull())
+        {
+            movement.Rooted = fields[4].GetBool();
+        }
+
+        if (!fields[5].IsNull())
+        {
+            movement.Chase = static_cast<CreatureChaseMovementType>(fields[5].GetUInt8());
+        }
+
+        if (!fields[6].IsNull())
+        {
+            movement.Random = static_cast<CreatureRandomMovementType>(fields[6].GetUInt8());
+        }
+
+        if (!fields[7].IsNull())
+        {
+            movement.InteractionPauseTimer = fields[7].GetUInt32();
```

Comment:
It's never implemented inthe core.

Reply:
i am looking into it now, but yeah, definately seems like i missed this one in particular.

Reply:
implemented. thanks again @UltraNix